kube-app-testing.sh

```diff
@@ -520,106 +432,105 @@ update_aws_sec_group () {
-    err "Nodepool creation failed."
-    exit 3
+  # align config with kind clusters
+  if [[ ! -d ${CONFIG_DIR} ]]; then
+    mkdir ${CONFIG_DIR}
   fi
 
-  info "Adding labels to cluster ${CLUSTER_ID}"
-  # label the cluster with some useful information. failure to label a cluster
-  # doesn't cause a job failure
-  if ! curl ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/labels/ -X PUT \
-      -H "Content-Type: application/json" \
-      -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-      -d "$(gen_gs_blob addlabels)" ; then
-    err "Could not label cluster, however the job will continue."
-  fi
+  info "Sleeping 10 seconds before create kubeconfig attempt"
+  sleep 10
 
-  # wait for the cluster to be ready
   # declare a counter
   _counter=0
-  until [ `curl -s -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/ | jq .conditions | grep -i "created" | wc -l` -gt 0 ]; do
-    # exit if the cluster hasn't been created in 30 minutes
+  info "Writing kubeconfig into ${KUBECONFIG}"
+  until gsctl create kubeconfig \
+    --cluster="${CLUSTER_ID}" \
+    --certificate-organizations=system:masters \
+    --force \
+    --self-contained="${KUBECONFIG}"
+  do
+    # exit if the kubeconfig hasn't been created in 30 minutes
     if [[ "$_counter" -gt 60 ]]; then
-      err "Cluster not created after 30 minutes."
+      err "Kubeconfig not created after 30 minutes."
       exit 3
     fi
 
     # increment the counter
     _counter=$((_counter+1))
-    info "Waiting for cluster ${CLUSTER_ID} to be ready."
+    info "Waiting for kubeconfig for cluster ${CLUSTER_ID} to be created."
     sleep 30
   done
 
-  info "Creating key-pair for tenant cluster access."
-  # create a key pair (must be stored directly in a variable)
-  _key_pair=$(curl ${GS_API_URL}/v4/clusters/${CLUSTER_ID}/key-pairs/ -X POST \
-    -H "Content-Type: application/json" \
-    -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-    -d "$(gen_gs_blob keypair)")
+  if [[ -z ${NO_EXTERNAL_KUBE_API} ]]; then
+    # update Security Group to allow access
+    update_aws_sec_group ${CLUSTER_ID}
 
-  # check that we actually got a key pair back
-  grep -q "certificate_authority_data" <<< $_key_pair
-  if [[ "$?" -gt 0 ]]; then
-    err "Key pair creation failed."
-    exit 3
-  fi
-
-  # parse required fields from key pair creation response in order to
-  # create a kubeconfig for the tenant cluster
-  TC_API_URI=$(curl -s -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/ | jq -r .api_endpoint)
-  CA_CERT=$(echo $_key_pair | jq -r '.certificate_authority_data | @base64')
-  CLIENT_CERT=$(echo $_key_pair | jq -r '.client_certificate_data | @base64')
-  CLIENT_KEY=$(echo $_key_pair | jq -r '.client_key_data | @base64')
-
-  # align config with kind clusters
-  if [[ ! -d ${CONFIG_DIR} ]]; then
-    mkdir ${CONFIG_DIR}
+    info "Sleeping for 30 seconds to ensure ingress rule has been applied."
+    sleep 30
```

Comment:
Again, sounds like a race condition. Should we check the existence of the ingress rule after the `sleep`?